subsys/mgmt/mcumgr/grp/enum_mgmt/src/enum_mgmt.c

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) - 2024 Extreme Engineering Solutions
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+#include <zephyr/debug/object_tracing.h>
+#include <zephyr/kernel_structs.h>
+#include <zephyr/mgmt/mcumgr/mgmt/mgmt.h>
+#include <zephyr/mgmt/mcumgr/smp/smp.h>
+#include <zephyr/mgmt/mcumgr/mgmt/handlers.h>
+#include <zephyr/mgmt/mcumgr/grp/enum_mgmt/enum_mgmt.h>
+#include <zephyr/logging/log.h>
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <zcbor_common.h>
+#include <zcbor_encode.h>
+#include <zcbor_decode.h>
+
+#include <mgmt/mcumgr/util/zcbor_bulk.h>
+
+static int enumeration_mgmt_list(struct smp_streamer *ctxt)
+{
+	int count;
+	zcbor_state_t *zse = ctxt->writer->zs;
+	bool ok;
+
+	count = mgmt_get_num_groups();
+	int ids[count];
+
+	count = mgmt_get_group_ids(ids);
+
+	if (count == 0) {
+		return MGMT_ERR_ENOTSUP;
```

Comment:
why does 0 mean not supported?

Reply:
Good point, it does not. I will correct that.